scrapy/pipelines/files.py

```diff
@@ -417,7 +417,14 @@ def _warn():
         ## end of deprecation warning block
 
         media_guid = hashlib.sha1(to_bytes(url)).hexdigest()  # change to request.url after deprecation
-        media_ext = os.path.splitext(url)[1]  # change to request.url after deprecation
+        
+        media_ext = os.path.splitext(url)[1] # change to request.url after deprecation
+        if not media_ext[1:-1].isalnum():
```

Comment:
What's the reason for testing on `[1:-1]` only?

Reply:
Hey! Do you merge this if done right? Those filenames with query string is really annoying, need to add custom pipeline every time..